devito/operations/interpolators.py

```diff
@@ -347,10 +373,9 @@ def _inject(self, field, expr, implicit_dims=None):
         # summing temp that wouldn't allow collapsing
         implicit_dims = implicit_dims + tuple(r.parent for r in self._rdim)
 
-        variables = variables + list(fields)
-
         # List of indirection indices for all adjacent grid points
-        idx_subs, temps = self._interp_idx(variables, implicit_dims=implicit_dims)
+        idx_subs, temps = self._interp_idx(list(fields), implicit_dims=implicit_dims,
```

Comment:
why does it need to be casted to a list? it was a tuple already, so it feels pretty odd